modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue

```diff
@@ -213,6 +213,11 @@ export default {
           });
         }
       }
+    },
+    tabs() {
+      if (!this.currentTab) {
+        this.currentTab = this.tabs[0].name;
```

Comment:
This isn't known until later in the case of page settings due to the way the type of the page is determined.

Reply:
Do we not need to do a deep change watch? What if the tabs are the same, but the contents are different? I'm not 100% sure of what is found or not with default watching. We could also tie this to `docType` changing after mount.

Reply:
`tabs` is a computed property. So what will that actually watch for?

"Computed properties are cached based on their reactive dependencies."

Following the stack of computed properties accessed by the `tabs()` computed function all the way to the bottom, we get to `moduleOptions`, which depends on the data property `this.docType`, which starts out correct for a piece type but as `@apostrophecms/page` for a page type. However it then changes dynamically based on newInstance or a change of `type`.

So since the correct set of tabs is fully determined by `type`, watching for that change of `type` is enough to do the job for us.

However, looking at this I did notice another scenario: the current tab can become invalid due to a switch of page type if the new page type does not have that tab. I added logic for that in a new commit.

Reply:
We're watching correctly — default watch behavior for computed properties is to run any time their "reactive dependencies" (other computed and data properties and props) change, and at the bottom of that stack is `moduleOptions`, which depends on `this.docType`, which is what we want.

However I did notice that the current tab can become invalid if you switch to a page type that doesn't have that tab, so I added logic for that too.